feature_apps/src/main/java/cm/aptoide/pt/feature_apps/data/AppsRepository.kt

```diff
@@ -0,0 +1,15 @@
+package cm.aptoide.pt.feature_apps.data
+
+import kotlinx.coroutines.flow.Flow
+
+interface AppsRepository {
+
+  fun getAppsList(url: String): Flow<AppsResult>
+
+}
+
+sealed interface AppsResult {
```

Comment:
Initially i was trying to use a Result<T> .
Should we follow that approach or this one ?

Reply:
We should use generic result, this will be the same for 99% of the classes.